src/mongo/db/pipeline/document_source_group.h

```diff
@@ -147,15 +147,21 @@ class DocumentSourceGroup final : public DocumentSource, public SplittableDocume
      */
     Value expandId(const Value& val);
 
+    struct Field {
```

Comment:
Let's call this `AccumulatedField`, to be more specific. `Field` is pretty generic.

Edit: we already have this struct: https://github.com/mongodb/mongo/blob/r3.5.3/src/mongo/db/pipeline/accumulation_statement.h! Let's just use that.